app/components/post/index.js

```diff
@@ -51,7 +55,13 @@ function mapStateToProps(state, ownProps) {
 
     const {deviceWidth} = getDimensions(state);
 
+    let channelIsReadOnly = false;
+    if (currentChannel.name === General.DEFAULT_CHANNEL) {
+        channelIsReadOnly = config.ExperimentalTownSquareIsReadOnly === 'true' && !(isAdmin(roles) && !isSystemAdmin(roles) && !isChannelAdmin(roles));
```

Comment:
Like I mentioned for the web app, this should in a selector

Reply:
Looks like I'm waiting on this to be merged https://github.com/mattermost/mattermost-redux/pull/429

Reply:
Looks like I'm waiting on this be merged https://github.com/mattermost/mattermost-redux/pull/429